testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,543 @@
-import numpy as np
+from MDAnalysisTests.datafiles import TRR_multi_frame, TRR_single_frame
+from MDAnalysisTests.datafiles import XTC_multi_frame, XTC_single_frame
+
+import pytest
+
+XTC_OFFSETS = np.array([0, 104, 208, 312, 416, 520, 624, 728, 832, 936])
+TRR_OFFSETS = np.array([0, 480, 960, 1440, 1920, 2400, 2880, 3360, 3840, 4320])
+
+
+@pytest.fixture
+def frame_xtc():
+    with XTCFile(XTC_single_frame) as f:
+        frame = f.read()
+        yield frame
+
+
+@pytest.fixture
+def xtc():
+    with XTCFile(XTC_multi_frame) as f:
+        yield f
+
+
+@pytest.fixture
+def frame_trr():
+    with TRRFile(TRR_single_frame) as f:
+        frame = f.read()
+        yield frame
+
+
+@pytest.fixture
+def trr():
+    with TRRFile(TRR_multi_frame) as f:
+        yield f
+
+
+@pytest.mark.parametrize('fname, xdr', ((XTC_single_frame, XTCFile),
+                                        (TRR_single_frame, TRRFile)))
+def test_natoms(fname, xdr):
+    with xdr(fname) as f:
+        assert f.n_atoms == 10
+
+
+def test_xtc_xyz(frame_xtc):
+    assert_array_almost_equal(frame_xtc.x.flat, np.arange(30), decimal=3)
+
+
+def test_trr_xyz(frame_trr):
+    assert_array_almost_equal(frame_trr.x.flat, np.arange(30))
+
+
+def test_trr_velocity(frame_trr):
+    assert_array_almost_equal(frame_trr.v.flat, np.arange(30))
+
+
+def test_trr_forces(frame_trr):
+    assert_array_almost_equal(frame_trr.f.flat, np.arange(30))
+
+
+def test_xtc_box(frame_xtc):
+    assert_array_almost_equal(frame_xtc.box, np.eye(3) * 20, decimal=3)
+
+
+def test_trr_box(frame_trr):
+    assert_array_almost_equal(frame_trr.box, np.eye(3) * 20)
+
+
+def test_xtc_step(frame_xtc):
+    assert frame_xtc.step == 0
 
 
-class XDRFormatBaseTest(object):
+def test_trr_step(frame_trr):
+    assert frame_trr.step == 10
 
-    def test_n_atoms(self):
-        f = self.xdrfile(self.single_frame)
-        assert_equal(f.n_atoms, 10)
 
-    @raises(IOError)
-    def test_raise_not_existing(self):
-        self.xdrfile('foo')
+def test_xtc_time(frame_xtc):
+    assert frame_xtc.time == 10.0
 
-    @raises(IOError)
-    def test_open_wrong_mode(self):
-        self.xdrfile('foo', 'e')
 
-    @raises(IOError)
-    def test_over_seek(self):
-        with self.xdrfile(self.multi_frame) as f:
+def test_trr_time(frame_trr):
+    assert frame_trr.time == 5.0
+
+
+def test_xtc_prec(frame_xtc):
+    assert frame_xtc.prec == 1000.0
+
+
+def test_trr_lmbda(frame_trr):
+    assert frame_trr.lmbda == .5
+
+
+@pytest.mark.parametrize("xdrfile", (XTCFile, TRRFile))
+def test_raise_not_existing(xdrfile):
+    with pytest.raises(IOError):
+        xdrfile('foo')
+
+
+@pytest.mark.parametrize("xdrfile", (XTCFile, TRRFile))
+def test_open_wrong_mode(xdrfile):
+    with pytest.raises(IOError):
+        xdrfile('foo', 'e')
+
+
+@pytest.mark.parametrize("xdrfile, fname", ((XTCFile, XTC_multi_frame),
+                                            (TRRFile, TRR_multi_frame)))
+def test_over_seek(xdrfile, fname):
```

Comment:
I'm unable in this test to use the `xtc` and `trr` fixtures above with parametrize.